source/wasm/src/project_system.rs

```diff
@@ -430,15 +445,14 @@ pub(crate) fn into_qsc_args(
     ),
     Vec<qsc::compile::Error>,
 > {
-    let capabilities = qsc::target::Profile::from_str(&program.profile())
-        .unwrap_or_else(|()| panic!("Invalid target : {}", program.profile()))
-        .into();
-
     let pkg_graph: PackageGraphSources = program.packageGraphSources().into();
     let pkg_graph: qsc_project::PackageGraphSources = pkg_graph.into();
 
-    // this function call builds all dependencies as a part of preparing the package store
-    // for building the user code.
+    // Use the project-level target_profile from ProgramConfig
+    let profile = qsc::target::Profile::from_str(&program.profile())
+        .unwrap_or_else(|()| panic!("Invalid target : {}", program.profile()));
```

Comment:
Why does this `panic` instead of returning `Err`?

Reply:
There shouldn't be a way for there to be an invalid profile in the program at this point. If this panic happens, it reflects something wrong with our code, not with something the user did. My understanding is that we only create and return formal errors for issues caused by the user doing something wrong, and that we use panics for issues caused compiler logic.